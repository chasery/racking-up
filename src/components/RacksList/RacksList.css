.RacksList {
  display: grid;
  grid-template-columns: 1fr;
  column-gap: 1rem;
  row-gap: 1rem;
  margin: 1rem 0 0;
  padding: 0;
}
.RacksList > li {
  align-self: start;
}

.RacksList__add {
  display: flex;
  flex-grow: 0;
  list-style: none;
  box-shadow: 0 0.0625rem 0.0625rem #5e43521c;
  border-radius: 0.125rem;
  overflow: hidden;
}
.RacksList__add a {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-grow: 1;
  padding: 1rem 1.5rem;
  background-color: #ead7dd;
  font-weight: 500;
  text-align: center;
  text-decoration: none;
}
.RacksList__add a:hover {
  background-color: #dcbcc6;
  color: #5e4352;
}
.RacksList__add a > img {
  margin-right: 0.5rem;
}

@media all and (min-width: 50rem) {
  .RacksList {
    grid-template-columns: repeat(2, 1fr);
  }
}
@media all and (min-width: 64rem) {
  .RacksList {
    grid-template-columns: repeat(3, 1fr);
  }
}
