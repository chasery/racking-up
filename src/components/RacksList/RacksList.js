import React from 'react';
import { Link } from 'react-router-dom';
import Rack from '../Rack/Rack';
import add from '../../assets/svg/add.svg';
import './RacksList.css';

function RacksList(props) {
  const renderRacks = (racks) => {
    if (racks) {
      return racks.map((rack) => (
        <Rack
          key={rack.rack_id}
          id={rack.rack_id}
          name={rack.rack_name}
          items={rack.items}
        />
      ));
    }
  };

  return (
    <ol className='RacksList'>
      {renderRacks(props.racks)}
      <li className='RacksList__add'>
        <Link to='/racks/add-rack'>
          <img src={add} alt='Add rack' />
          <span>Rack</span>
        </Link>
      </li>
    </ol>
  );
}

export default RacksList;
