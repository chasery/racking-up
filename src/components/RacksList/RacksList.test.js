import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import RacksList from "./RacksList";

describe("RacksList component", () => {
  it("renders without error", () => {
    const wrapper = shallow(<RacksList />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
