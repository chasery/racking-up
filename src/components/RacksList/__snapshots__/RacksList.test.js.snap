// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RacksList component renders without error 1`] = `
<ol
  className="RacksList"
>
  <Rack
    id={1}
    key="1"
    name="Fall 2021"
  />
  <Rack
    id={2}
    key="2"
    name="Summer 2021"
  />
  <Rack
    id={3}
    key="3"
    name="Anniversary Outfit"
  />
  <li
    className="RacksList__add"
  >
    + Add Rack
  </li>
</ol>
`;
